src/emu-i386/simx86/sigsegv.c

```diff
@@ -361,6 +340,24 @@ int e_emu_fault(struct sigcontext *scp)
 	return TryMemRef;
   }
 #endif
+
+  /* CONFIG_CPUSIM only: */
+  if (_trapno==0x0e) {
+	if (in_dpmi_emu) {
+		/* reflect DPMI page fault back to a DOSEMU crash or
+		   DPMI exception;
+		   vm86 faults will terminate DOSEMU via "return 0"
+		*/
+		TheCPU.err = EXCP0E_PAGE;
+		TheCPU.scp_err = _err;
+		TheCPU.cr2 = _cr2;
+		TheCPU.eip = P0 - LONG_CS;
+		fault_cnt--;
+		siglongjmp(jmp_env, 0);
```

Comment:
sigaltstack is not needed at all for this case (only with $_cpu_emu="fullsim"), and neither for "full" nor KVM+DPMI for that matter.
So it would be easiest to just remove SA_ONSTACK if native DPMI is not used.